h2o-automl/src/main/java/ai/h2o/automl/targetencoding/TargetEncoder.java

```diff
@@ -520,27 +565,30 @@ public void map(Chunk cs[]) {
      * @param foldColumnName column's name that contains fold number the row is belong to.
      * @param withBlendedAvg whether to apply blending or not.
      * @param noiseLevel amount of noise to add to the final encodings.
-     * @param imputeNAsWithNewCategory set to `true` to impute NAs with new category.
      * @param seed we might want to specify particular values for reproducibility in tests.
      * @return copy of the `data` frame with encodings
      */
     public Frame applyTargetEncoding(Frame data,
                                      String targetColumnName,
                                      Map<String, Frame> columnToEncodingMap,
-                                     byte dataLeakageHandlingStrategy,
+                                     DataLeakageHandlingStrategy dataLeakageHandlingStrategy,
                                      String foldColumnName,
                                      boolean withBlendedAvg,
                                      double noiseLevel,
-                                     boolean imputeNAsWithNewCategory,
-                                     long seed) {
+                                     long seed,
+                                     Key<Frame> encodedFrameKey) {
 
         if(noiseLevel < 0 )
             throw new IllegalStateException("`_noiseLevel` must be non-negative");
 
         Frame dataWithAllEncodings = null;
         try {
+          
+          if(encodedFrameKey == null){
+            encodedFrameKey = Key.make();
```

Comment:
But we do have support for no encodedFrameKey provided - in that case, we'll just make up our own !